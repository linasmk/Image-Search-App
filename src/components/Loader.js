/* ========= App Dependencies ============= */
import React from "react";
import PropTypes from "prop-types";
/* ========= Code ============= */
export const Loader = ({ bottomLoader, display }) => (
  <div className="loader-box">
    <div
      style={bottomLoader ? { display: "block" } : { display: display }}
      className="loader"
    />
  </div>
);
Loader.defaultProps = {
  bottomLoader: undefined,
  display: undefined
};
Loader.propTypes = {
  bottomLoader: PropTypes.bool,
  display: PropTypes.string
};
export default Loader;
