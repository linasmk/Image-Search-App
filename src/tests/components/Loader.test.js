/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
/* ========= Components ============= */
import { Loader } from "@comp/Loader";
/* ========= Code ============= */
let wrapper;

beforeEach(() => {
  wrapper = shallow(<Loader />);
});

it("renders Loader component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
