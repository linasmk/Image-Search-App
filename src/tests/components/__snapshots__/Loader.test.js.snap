// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader component correctly 1`] = `
<div
  className="loader-box"
>
  <div
    className="loader"
    style={
      Object {
        "display": undefined,
      }
    }
  />
</div>
`;
